src/staticarrays_support.jl

```diff
@@ -27,3 +27,58 @@ StructArrays.component(s::StaticArray, i) = getindex(s, i)
 end
 StructArrays.component(s::FieldArray, i) = invoke(StructArrays.component, Tuple{Any, Any}, s, i)
 StructArrays.createinstance(T::Type{<:FieldArray}, args...) = invoke(createinstance, Tuple{Type{<:Any}, Vararg}, T, args...)
+
+# Broadcast overload
+using StaticArraysCore: StaticArrayStyle, similar_type
+StructStaticArrayStyle{N} = StructArrayStyle{StaticArrayStyle{N}, N}
+function Broadcast.instantiate(bc::Broadcasted{StructStaticArrayStyle{M}}) where {M}
+    bc′ = Broadcast.instantiate(replace_structarray(bc))
+    return convert(Broadcasted{StructStaticArrayStyle{M}}, bc′)
+end
+# This looks costy, but compiler should be able to optimize them away
```

Comment:
```suggestion
# This looks costly, but the compiler should be able to optimize them away
```

Can we test this somehow? Is the issue that one needs to figure out whether the StaticArrays custom axes should be used?

Reply:
This is called only for `StructStaticArrayStyle`. So other `BroadcastStyle`s wont use this routine.
I have test it via `@test @inferred(Broadcast.axes(bc)) === axes(s)`.
This test should be enough as the result is a `Tuple` of `SOneTo`.